jupyter-lib/shared-compiler/src/main/kotlin/org/jetbrains/kotlinx/jupyter/repl/creating/LazilyConstructibleReplComponentsProvider.kt

```diff
@@ -29,8 +30,10 @@ abstract class LazilyConstructibleReplComponentsProvider : ReplComponentsProvide
     override val homeDir: File? by lazy { provideHomeDir() }
     protected abstract fun provideHomeDir(): File?
 
-    override val debugPort: Int? by lazy { provideDebugPort() }
-    protected abstract fun provideDebugPort(): Int?
+    override val debugPort: Int? by lazy { debugIntegrationProvider.provideDebugPort() }
```

Comment:
No, that's not how this pattern works. You can remove `val debugPort` completely if you wish. But just observe the code: is there any lazy implementation other than `lazy { provide*() }`?

Reply:
I agree, that's a violation of the pattern. Probably, we might need debug port for future cases in embedded kernel